Add SOMns applications and tests

ctrlpz opened the pull request:
- add five SOMns applications and tests. These applications feature asynchronous message passing using actors. The goal is to use them later for a user study to evaluate the debugging features for actors.

ctrlpz committed:
Add SOMns applications and tests

- add five SOMns applications and tests. These applications feature asynchronous message passing using actors. The goal is to use them later for a user study to evaluate the debugging features for actors.

ctrlpz committed:
Update shopping application program

- remove timesRepeat in the run method of Listener class and remove the timeout in the main, they are not necessary
- update test

smarr commented on core-lib/demos/applications/InstantMessengerApplication.ns:
```diff
@@ -0,0 +1,165 @@
+    )
+)
+
+ class TextMessage new: content sender: senderName = TransferObject (
```
Hm, is there a specific reason why this is a TransferObject?
Seems like the slots are immutable, so, it looks like this could be a `Value`?

smarr replied:
Looking at the tests, this seems indeed be a value object. So, I changed it.

smarr commented on core-lib/demos/applications/InstantMessengerApplication.ns:
```diff
@@ -0,0 +1,165 @@
+)
+
+ class TextMessage new: content sender: senderName = TransferObject (
+  | private msg = content.
```
Since these slots are immutable, it seems unnecessary to have getters (`getContent`, `getSender`). Instead, these slots could simply be made `public`.

smarr replied:
fixed

smarr commented on core-lib/demos/applications/InstantMessengerApplication.ns:
```diff
@@ -0,0 +1,165 @@
+                pSend whenResolved:[:result |
+                    (result = 'ok')
+                    ifTrue:[
+                         array:: TransferArray new: 2. (* array with the msg to display in the sender actor screen *)
```
You got a `TextMessage` class here, which seems to fulfill the same purpose. Would it make sense to use it instead?

smarr replied:
Fixed

smarr commented on core-lib/demos/applications/InstantMessengerApplication.ns:
```diff
@@ -0,0 +1,165 @@
+        ^ pp promise
+    )
+
+    public getName = (
```
`name` is immutable. The getter seems unnecessary.

smarr replied:
Fixed.

smarr commented on core-lib/demos/applications/InstantMessengerApplication.ns:
```diff
@@ -0,0 +1,165 @@
+                pSend:: sendMessage: remoteName contentMsg: msg.
+
+                pSend whenResolved:[:result |
+                    (result = 'ok')
```
Instead of using strings, in Smalltalk, it would be more idiomatic to use a Symbol: `#ok`.

smarr replied:
Fixed.

smarr commented on core-lib/demos/applications/InstantMessengerApplication.ns:
```diff
@@ -0,0 +1,165 @@
+      self displayMessage:  sender +': '+ content.
+
+      (* acknowledgement notification of message received *)
+      ^ 'ok'.
```
The earlier comment about the Symbol, would apply here as well.

smarr replied:
Fixed.

smarr committed:
Whitespace InstantMessengerApplication.ns

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr reviewed (commented):
Just looking at the first app

smarr committed:
Fix bug in AsyncTestCase: misnamed ex variable

Signed-off-by: Stefan Marr <git@stefan-marr.de>

smarr committed:
Make InstantMessangerApplication more idiomatic

Signed-off-by: Stefan Marr <git@stefan-marr.de>

ctrlpz committed:
Minor changes

- use symbols instead of strings
- minor code formatting and remove some whitespaces and empty lines
- remove unnecessary new keyword in ShoppingApplication, as an unnecessary eventual send
- update changes in tests
